web/src/engine/websites/ProjectSuki.ts

```diff
@@ -0,0 +1,72 @@
+import { Tags } from '../Tags';
+import icon from './ProjectSuki.webp';
+import { DecoratableMangaScraper, Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchWindowScript } from '../platform/FetchProvider';
+
+const pageScript = `
+    new Promise ((resolve, reject) => {
+        try {
+
+            const element = document.querySelector('.strip-reader');
+            const bookid = window.location.href.split('/') [4];
+            const chapterid = window.location.href.split('/') [5];
+            const body = JSON.stringify({
+                bookid: bookid,
+                chapterid: chapterid,
+                first: true
+            });
+
+            const xhr = new XMLHttpRequest();
+            xhr.onload = function () {
+                if (xhr.status >= 200 && xhr.status < 300) {
```

Comment:
Use `try ... catch` in asynchronous function and properly reject the promise to avoid uncaught exceptions which will lead to the promise never being fulfilled.